src/hooks/useScreenWidth.ts

```diff
@@ -0,0 +1,21 @@
+import { useState, useEffect } from "react"
+
+const useScreenWidth = () => {
+  const [width, setWidth] = useState<number>(window.innerWidth)
+
+  useEffect(() => {
+    const handleResize = () => {
+      setWidth(window.innerWidth)
+    }
+
+    window.addEventListener("resize", handleResize)
```

Comment:
Potentially could be throttled a bit, or too many setState happen.

Reply:
We can do this but I don't think that screen would be resized frequently, it is more likely edge cases.